src/client/sky.cpp

```diff
@@ -700,69 +570,198 @@ void Sky::update(float time_of_day, float time_brightness,
 			if (time_brightness < 0.5f)
-				cloud_direct_brightness = std::fmax(cloud_direct_brightness,
-					time_brightness * 1.3f);
+				cloud_direct_brightness =
+						std::fmax(cloud_direct_brightness,
+								time_brightness * 1.3f);
 		}
 	} else {
 		cloud_direct_brightness = direct_brightness;
 	}
 
-	m_cloud_brightness = m_cloud_brightness * cloud_color_change_fraction +
-		cloud_direct_brightness * (1.0 - cloud_color_change_fraction);
-	m_cloudcolor_f = video::SColorf(
-		m_cloudcolor_bright_f.r * m_cloud_brightness,
-		m_cloudcolor_bright_f.g * m_cloud_brightness,
-		m_cloudcolor_bright_f.b * m_cloud_brightness,
-		1.0
-	);
+	m_cloud_brightness =
+			m_cloud_brightness * cloud_color_change_fraction +
+			cloud_direct_brightness * (1.0 - cloud_color_change_fraction);
+	m_cloudcolor_f = video::SColorf(m_cloudcolor_bright_f.r * m_cloud_brightness,
+			m_cloudcolor_bright_f.g * m_cloud_brightness,
+			m_cloudcolor_bright_f.b * m_cloud_brightness, 1.0);
 	if (m_directional_colored_fog) {
-		m_cloudcolor_f = m_mix_scolorf(m_cloudcolor_f,
-			video::SColorf(pointcolor), m_horizon_blend() * 0.25);
+		m_cloudcolor_f = m_mix_scolorf(m_cloudcolor_f, video::SColorf(pointcolor),
+				m_horizon_blend() * 0.25);
+	}
+}
+
+void Sky::draw_sun(video::IVideoDriver *driver, float sunsize, video::SColor suncolor,
+		video::SColor suncolor2, float wicked_time_of_day)
+{
+	static const u16 indices[4] = {0, 1, 2, 3};
+	std::array<video::S3DVertex, 4> vertices;
+	if (!m_sun_texture) {
+		driver->setMaterial(m_materials[1]);
+		const float sunsizes[4] = {
+				sunsize * 1.7f, sunsize * 1.2f, sunsize, sunsize * 0.7f};
+		video::SColor c1 = suncolor;
+		video::SColor c2 = suncolor;
+		c1.setAlpha(0.05 * 255);
+		c2.setAlpha(0.15 * 255);
+		const video::SColor colors[4] = {c1, c2, suncolor, suncolor2};
+		for (int i = 0; i < 4; i++) {
+			vertices = draw_sky_body(-sunsizes[i], sunsizes[i], colors[i]);
+			vertices = place_sky_body(
+					vertices, 90, wicked_time_of_day * 360 - 90);
+			driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+		}
+	} else {
+		driver->setMaterial(m_materials[3]);
+		float d = sunsize * 1.7;
+		video::SColor c;
+		if (m_sun_tonemap)
+			c = video::SColor(0, 0, 0, 0);
+		else
+			c = video::SColor(255, 255, 255, 255);
+		vertices = draw_sky_body(-d, d, c);
+		vertices = place_sky_body(vertices, 90, wicked_time_of_day * 360 - 90);
+		driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+	}
+}
+
+void Sky::draw_moon(video::IVideoDriver *driver, float moonsize, video::SColor mooncolor,
+		video::SColor mooncolor2, float wicked_time_of_day)
+{
+	static const u16 indices[4] = {0, 1, 2, 3};
+	std::array<video::S3DVertex, 4> vertices;
+	if (!m_moon_texture) {
+		driver->setMaterial(m_materials[1]);
+		const float moonsizes_1[4] = {
+				-moonsize * 1.9f, -moonsize * 1.3f, -moonsize, -moonsize};
+		const float moonsizes_2[4] = {moonsize * 1.9f, moonsize * 1.3f, moonsize,
+				moonsize * 0.6f};
```

Comment:
Please only wrap lines at 90 columns (tab size 4). This may be a bad clang format suggestion.